### PR TITLE
Add some basic deployment docs

### PR DESCRIPTION
**Pull request by djmitche**
This is quite incomplete but I figured I'd get a start so we have something to improve on.

### CODE REVIEW

File: deployment-docs/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+Taskcluster is deployed using Terraform.
+The Terraform configuration is in `infrastructure/terraform` and consists of a Terraform module that should be included into your own Terraform configuration, providing the necessary configuration.
+See the `variables.tf` file for a list of required values.
+
+The Terraform module contains a [reference](../infrastructure/terraform/taskcluster.tf.json) to the Docker image that will be deployed.
+This is updated periodically in the repository; a more formal approach is in the design stages.
+
+## Clouds
+
+Taskcluster is a multi-cloud system, and as such expects credentials for several cloud providers.
```
Comment (imbstack) [inline deployment-docs/README.md]: This is a very positive way to spin this. :100:

Comment (djmitche) [reply deployment-docs/README.md]: "Taskcluster makes great lemonade, so it ships with a large quantity of lemons"

File: deployment-docs/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Deployment Documentation
+
+*NOTE*: This is a work in progress, as we develop processes for flexibly and reliably deploying Taskcluster.
+
+## Overview
+
+A Taskcluster deployment is identified by a rootUrl, which defines the domain name on which the deployment can be accessed.
```
Comment (helfi92) [inline deployment-docs/README.md]: s/rootUrl/`rootUrl`/

**Comment by djmitche** [comment]
@helfi92 I can't quite figure out what users should supply for `audience` and `scope` in the mozilla-auth0 configuration.  I thought they were unused, but it seems that's not true.  Can you suggest some text for that section?

**Comment by helfi92** [comment]
The `audience` should be set to `api.sso.mozilla.com` and the `scope` to `display:all taskcluster-credentials openid profile`.

https://github.com/mozilla-iam/cis/blob/00cebf88e83cfff68499af37f3109a4423c03dc8/docs/PersonAPI.md#full-list-of-scopes

**Pull request merged**